bitcoin/src/network/constants.rs

```diff
@@ -186,6 +215,25 @@ impl fmt::Display for Network {
     }
 }
 
+/// Error in parsing network from chain hash.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct UnknownChainHash(ChainHash);
+
+impl TryFrom<ChainHash> for Network {
+    type Error = UnknownChainHash;
+
+    fn try_from(chain_hash: ChainHash) -> Result<Self, Self::Error> {
+        match chain_hash {
+            // Note: any new network entries must be matched against here.
+            ChainHash::BITCOIN => Ok(Network::Bitcoin),
+            ChainHash::TESTNET => Ok(Network::Testnet),
+            ChainHash::SIGNET => Ok(Network::Signet),
+            ChainHash::REGTEST => Ok(Network::Regtest),
+            _ => Err(UnknownChainHash(chain_hash)),
+        }
+    }
+}
```

Comment:
This is great! Could you also impl `Display` and `std::error::Error` for `UnknownChainHash`? Suggested message: `"unknown chain hash {}"`.